llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -510,8 +511,11 @@ def forward(
             use_cache=use_cache,
         )
 
-        logits = F.linear(outputs.last_hidden_state,
-                          self.transformer.wte.weight)
+        # move outputs to same device as weights for token embedding
+        # needed to support HF `device_map`
+        logits = F.linear(
+            outputs.last_hidden_state.to(self.transformer.wte.weight.device),
```

Comment:
Just want to verify that this does not break training with FSDP.
`self.transformer.wte.weight` should be sharded, so `self.transformer.wte.weight.device` might not make sense
or is it that durring training, within the FSDP block, `self.transformer.wte.weight` is set to the current device and therefore `device` is correct?

Reply:
@vchiley I will do some test runs with FSDP on single node 8-GPUs. Would that satisfy your concern?

FWIW I think the `transformer.wte.weight.device` will return the expected value and the storage location (sharded across the GPUs) is hidden from the end user. Since we are within the `transformer` FSDP block, after the forward all-gather, things look "normal"

Reply:
Why not just try 2 node
run for 100 itrs and if its fine, I'm fine.

2 node tests all the things

Reply:
Confirmed that a 2-node job with MPT-1B and FSDP works with this branch.